subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultProjectDependencyConstraint.java

```diff
@@ -33,7 +33,7 @@
  * platforms.
  */
 public class DefaultProjectDependencyConstraint implements DependencyConstraintInternal {
-    private final ProjectDependency projectDependency;
+    public final ProjectDependency projectDependency;
```

Comment:
Hi,

Coincidentally, I have spent a good part of today hunting this down and fixing it. Looks like we implemented exactly the same fix independently!

The only difference in my case is that I thought it would be a good idea to extract an interface from `DefaultProjectDependencyConstraint`, so that `DefaultMavenPublication` does not depend on internal classes. And exposed `projectDependency` via a property getter instead of exposing it as field.

**ProjectDependencyConstraint.java**
```java
package org.gradle.api.artifacts;

/**
 * <p>A {@code ProjectDependencyConstraint} is a {@link DependencyConstraint} on another project in the current project hierarchy.</p>
 */
public interface ProjectDependencyConstraint extends DependencyConstraint {

    /**
     * Returns the project dependency associated with this project dependency constraint.
     */
    ProjectDependency getProjectDependency();
}
```

**DefaultProjectDependencyConstraint.java**
```java
public class DefaultProjectDependencyConstraint implements ProjectDependencyConstraint, DependencyConstraintInternal {

    @Override
    public ProjectDependency getProjectDependency() {
        return projectDependency;
    }
```